src/pyuvdata/uvbeam/analytic_beam.py

```diff
@@ -0,0 +1,893 @@
+    """
+    Find the sigma that gives a beam width similar to an Airy disk.
+
+    Find the stddev of a gaussian with fwhm equal to that of
+    an Airy disk's main lobe for a given diameter.
+
+    Parameters
+    ----------
+    diameter : float
+        Antenna diameter in meters
+    freq_array : array of float
+        Frequencies in Hz
+
+    Returns
+    -------
+    sigma : float
+        The standard deviation in zenith angle radians for a Gaussian beam
+        with FWHM equal to that of an Airy disk's main lobe for an aperture
+        with the given diameter.
+
+    """
+    wavelengths = speed_of_light.to("m/s").value / freq_array
+
+    scalar = 2.2150894  # Found by fitting a Gaussian to an Airy disk function
+
+    sigma = np.arcsin(scalar * wavelengths / (np.pi * diameter)) * 2 / 2.355
+
+    return sigma
+
+
+@dataclass
+class GaussianBeam(AnalyticBeam):
+    """
+    Define a Gaussian beam, optionally with frequency dependent size.
+
+    Parameters
+    ----------
+    sigma : float
+        Standard deviation in radians for the gaussian beam. Only one of sigma
+        and diameter should be set.
+    sigma_type : str
+        Either "efield" or "power" to indicate whether the sigma specifies the size of
+        the efield or power beam. Ignored if `sigma` is None.
+    diameter : float
+        Dish diameter in meters to use to define the size of the gaussian beam, by
+        matching the FWHM of the gaussian to the FWHM of an Airy disk. This will result
+        in a frequency dependent beam.  Only one of sigma and diameter should be set.
+    spectral_index : float
+        Option to scale the gaussian beam width as a power law with frequency. If set
+        to anything other than zero, the beam will be frequency dependent and the
+        `reference_frequency` must be set. Ignored if `sigma` is None.
+    reference_frequency : float
+        The reference frequency for the beam width power law, required if `sigma` is not
+        None and `spectral_index` is not zero. Ignored if `sigma` is None.
+    feed_array : np.ndarray of str
+        Feeds to define this beam for, e.g. n & e or x & y or r & l.
+    include_cross_pols : bool
+        Option to include the cross polarized beams (e.g. xy and yx or en and ne) for
+        the power beam.
+
+    """
+
+    sigma: float | None = None
+    sigma_type: Literal["efield", "power"] = "efield"
+    diameter: float | None = None
+    spectral_index: float = 0.0
+    reference_frequency: float = None
+
+    basis_vector_type = "az_za"
+
+    def __init__(
+        self,
+        *,
+        sigma: float | None = None,
+        sigma_type: Literal["efield", "power"] = "efield",
+        diameter: float | None = None,
+        spectral_index: float = 0.0,
+        reference_frequency: float = None,
+        feed_array: npt.NDArray[str] | None = None,
+        include_cross_pols: bool = True,
+    ):
```

Comment:
same comment above that you shouldn't need `__init__` (it is just redundant with the fields defined above). For validation and conversion of inputs, use `__post_init__`.